recipe/schemas/parsed_constructors.py

```diff
@@ -58,7 +58,108 @@ def stringify(value):
     return "if(" + ",".join(parts) + ")", "if(" + ",".join(order_by_parts) + ")"
 
 
-def create_ingredient_from_parsed(ingr_dict, builder, debug=False):
+def get_convert_dates(format: str):
+    """If a format only shows years or months, wrap dates in a conversion in the builder"""
+    if not isinstance(format, str):
+        return
+    if format == "%Y":
+        return "year_conv"
+    # A format for months but not days
+    elif (
+        "%B" in format and "%Y" in format and "%-d" not in format and "%d" not in format
+    ):
+        return "month_conv"
+
+
+def get_convert_datetimes(format: str):
+    """If a format only shows years or months, wrap datetimes in a conversion in the builder"""
+    if not isinstance(format, str):
+        return
+    if format == "%Y":
+        return "dt_year_conv"
+    # A format for months but not days
+    elif (
+        "%B" in format and "%Y" in format and "%-d" not in format and "%d" not in format
+    ):
+        return "dt_month_conv"
+    elif "%H" not in format:
+        return "dt_day_conv"
```

Comment:
This is a change; it's a slight improvement on how this worked before but hardcoding these formats is still yuck.